### ISSUE TITLE
Google Calendar Block: block stops displaying when page turns to draft

### ISSUE THREAD

**Issue by alaczek** [Automattic/jetpack#14825]
This might be a bit of an edge case, but Google Calendar block added to hompepage stopped displaying in both editor and on the frontend when I switched themes. This happened only with Varia child themes, so might have to do with turning current homepage into a draft on theme switch.

#### Steps to reproduce the issue

1. Activate one of Varia child themes (for example Maywood)
2. Add Google Calendar block to the homepage.
3. Switch themes to a different Varia child theme (for example Barnsbury)
4. Naviagte to My Site > Pages > Drafts (previous theme homepage turns into a draft on theme switch) 
5. Edit the old homepage. Notice that Google Calendar block is not visible, although there’s an empty space in its place. If you click on it, you’ll see the block controls, and the embed code is in the settings in the sidebar. 
6. If at this point you switch back to the original theme, the issue with Google Calendar persists (even though the homepage status automatically returns to published from draft).
7. I tried the above steps with Calendly block, and it did not disappear on theme switch, when the homepage turned to draft.

Video of the issue: https://cloudup.com/c7GMeS-o7k8

**Comment by apeatling** [comment]
@obenland Watching the video, I think this theme change should have switched homepage templates but it didn't. Any thoughts on what might have happened there?

**Comment by obenland** [comment]
It looks like the Dalston homepage template is in draft as I'd expect?

![Screen Shot 2020-02-28 at 9 12 48 AM](https://user-images.githubusercontent.com/1398304/75569776-a0131780-5a0a-11ea-97b4-ed67e135e9c1.png)

**Comment by apeatling** [comment]
Yes, my mistake, the video is looking at the same homepage template in draft. 🤦‍♂

**Comment by apeatling** [comment]
This seems to be caused by:

```
Refused to display 'https://calendar.google.com/calendar/embed?src=tl4njqffodltemv385vnifrjadm345g2%40import.calendar.google.comu0026amp;ctz=America%2FVancouver' in a frame because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'self'".
```

**Comment by apeatling** [comment]
Republishing the page does not fix this issue.

**Comment by glendaviesnz** [comment]
FYI - an additional detail - this doesn't seem to happen on a local wp install, only on wpcom.

**Comment by glendaviesnz** [comment]
For some reason Google is returning different response headers in this instance. Normally the embed http request gets just the following content security header:

`content-security-policy-report-only: script-src 'report-sample' 'nonce-5pueUMrCmKMiD2lYn/zUeQ' 'unsafe-inline' 'strict-dynamic' https: http: 'unsafe-eval';object-src 'none';base-uri 'self';report-uri /calendar/cspreport`

but in this instance it returns
`content-security-policy: frame-ancestors 'self'`

Just having a look to try and work out what causes Google to return these different headers.

**Comment by glendaviesnz** [comment]
Looks like a content encoding issue, for some reason when converted to draft some of the embed url entities get messed up: 

`https://calendar.google.com/calendar/embed?src=a8c.com_hfen27115pqgvfuf9n2mue943c%40group.calendar.google.com**\u0026amp**;ctz=America%2FChicago`

becomes

`https://calendar.google.com/calendar/embed?src=a8c.com_hfen27115pqgvfuf9n2mue943c%40group.calendar.google.com**u0026amp;**ctz=America%2FChicago`

Adding back the missing \ fixes it again.

**Comment by apeatling** [comment]
This seems to be done by the serializer here:

https://github.com/WordPress/gutenberg/blob/4857ad58c1241b3d63d21a6880c989b85746c3dc/packages/blocks/src/api/serializer.js#L217

**Comment by apeatling** [comment]
Somewhere in the dotcom codebase, the slash is being stripped out. The question is where. :)

**Comment by pablinos** [comment]
My guess is that it's related to [`wp_kses` being applied to attribute values](https://github.com/WordPress/WordPress/blob/master/wp-includes/blocks.php#L334). I've had problems with this encoding and stripping characters from attribute values on a local WordPress install when a user doesn't have unfiltered-html privileges.

**Comment by creativecoder** [comment]
This one is challenging to track down and fix--it's on @Automattic/explorers list, but we're keeping it in the backlog for now while we focus on higher priority issues.

**Comment by yansern** [comment]
Just want to add that this is not replicable when changing the post to draft from the editor.
It is only replicable when setting a draft from a theme switch.
